Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSmearing.cpp

```diff
@@ -127,20 +127,29 @@ ActsExamples::ProcessCode ActsExamples::ParticleSmearing::execute(
 
       // build the track covariance matrix using the smearing sigmas
       Acts::BoundSquareMatrix cov = Acts::BoundSquareMatrix::Zero();
-      cov(Acts::eBoundLoc0, Acts::eBoundLoc0) =
-          m_cfg.initialVarInflation[Acts::eBoundLoc0] * sigmaD0 * sigmaD0;
-      cov(Acts::eBoundLoc1, Acts::eBoundLoc1) =
-          m_cfg.initialVarInflation[Acts::eBoundLoc1] * sigmaZ0 * sigmaZ0;
-      cov(Acts::eBoundTime, Acts::eBoundTime) =
-          m_cfg.initialVarInflation[Acts::eBoundTime] * sigmaT0 * sigmaT0;
-      cov(Acts::eBoundPhi, Acts::eBoundPhi) =
-          m_cfg.initialVarInflation[Acts::eBoundPhi] * sigmaPhi * sigmaPhi;
-      cov(Acts::eBoundTheta, Acts::eBoundTheta) =
-          m_cfg.initialVarInflation[Acts::eBoundTheta] * sigmaTheta *
-          sigmaTheta;
-      cov(Acts::eBoundQOverP, Acts::eBoundQOverP) =
-          m_cfg.initialVarInflation[Acts::eBoundQOverP] * sigmaQOverP *
-          sigmaQOverP;
+      if (m_cfg.initialSigmas) {
+        // use the initial sigmas if set
+        for (std::size_t i = Acts::eBoundLoc0; i < Acts::eBoundSize; ++i) {
+          cov(i, i) = m_cfg.initialVarInflation[i] * (*m_cfg.initialSigmas)[i] *
+                      (*m_cfg.initialSigmas)[i];
+        }
```

Comment:
Same here, I think this would work:
```suggestion
        cov.diagonal() =  m_cfg.initialVarInflation * (*m_cfg.initialSigmas) *
                                     (*m_cfg.initialSigmas);
```